frontend/react/style-guide.md

```diff
@@ -925,6 +926,114 @@ src
   const WrappedComponent = composedHoc(Component);
   ```
 
+## Typescript
+
+- Useful links
+  - https://github.com/typescript-cheatsheets/react-typescript-cheatsheet
+  - https://www.typescriptlang.org/docs/home.html
+
+- File extensions
+  - Use `.tsx` extension for React components.
+  - Use `.ts` extension for files that don't need react elements
+
+- Creating a class component
+  ```tsx
+    interface Props {
+      prop1: string; //required prop
+      prop2?: number; //optional prop
+    }
+
+    interface State {
+      stateProperty: boolean;
+    }
+
+    class Breadcrumb extends Component<Props, State> {
+      //
+    }
+  ```
+
+- Creating a functional component
+  ```tsx
+    interface Props {
+      prop1: string;
+    }
+
+    function CardLink({ prop1, prop2}: Props) {
+      //
```

Comment:
what's with the empty comments?

Reply:
They are there just instead of the real code, we can use:
```js
 // ...
```
like in other examples.